Source/Core/Core/Core.h

```diff
@@ -159,7 +158,6 @@ void FrameUpdateOnCPUThread();
 void OnFrameEnd(Core::System& system);
 
 // Run a function on the CPU thread, asynchronously.
-// This is only valid to call from the host thread, since it uses PauseAndLock() internally.
 void RunOnCPUThread(Core::System& system, Common::MoveOnlyFunction<void()> function,
```

Comment:
`RunOnCPUThread` uses `Host_YieldToUI` which does
```
qApp->processEvents(QEventLoop::ExcludeUserInputEvents);
```
which I think needs to come from the proper thread, so at least with Qt I think calls still need to come from the host thread (or CPU thread which uses the fast path).

Reply:
Well, only when `wait_for_completion = true`, which arguably could be removed entirely since `CPUThreadGuard` exists.

After #13594 I think there is only one instance of `wait_for_completion = true` (in Movie.cpp) which could be easily changed to use `CPUThreadGuard`.